src/main/java/uk/gov/hmcts/cmc/claimstore/events/ccd/CCDCaseHandler.java

```diff
@@ -49,7 +48,7 @@ public CCDCaseHandler(
         this.userService = userService;
     }
 
-    @TransactionalEventListener
```

Comment:
The async nature of these calls means that there's no guarantee that the save does not happen before the document upload. This is one way around it. The other option is to tell Spring give priority to this call using the Order annotation and give it the highest priority.